### PR TITLE
Added register and return target code offset opencover options

### PR DESCRIPTION
**Pull request by phenixdotnet**
Hi.

A small PR to add 2 options on OpenCover settings and runner.

Regards

### CODE REVIEW

File: src/Cake.Common/Tools/OpenCover/OpenCoverRunner.cs [hunk 1]
```diff
@@ -121,8 +121,20 @@ public OpenCoverRunner(IFileSystem fileSystem, ICakeEnvironment environment, IPr
                 builder.AppendSwitch("-excludebyfile", ":", filters.Quote());
             }
 
-            // Use per-user registration of code coverage profiler.
-            builder.AppendSwitch("-register", ":", "user");
+            if (string.IsNullOrEmpty(settings.Register))
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/OpenCover/OpenCoverRunner.cs]: Maybe it's better to set the default value of `settings.Register` to `user`.

File: src/Cake.Common/Tools/OpenCover/OpenCoverSettings.cs [hunk 2]
```diff
@@ -44,6 +44,24 @@ public ISet<string> ExcludedFileFilters
         }
 
         /// <summary>
+        /// Gets or sets the register option
+        /// </summary>
+        public string Register
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/OpenCover/OpenCoverSettings.cs]: Should be `public string Register { get; set; }` to be more like other aliases.

File: src/Cake.Common/Tools/OpenCover/OpenCoverSettings.cs [hunk 3]
```diff
@@ -44,6 +44,24 @@ public ISet<string> ExcludedFileFilters
...
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Gets or sets the Return target code offset to be used
+        /// </summary>
+        public int? ReturnTargetCodeOffset
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/OpenCover/OpenCoverSettings.cs]: Should be `public int? ReturnTargetCodeOffset { get; set; }` to be more like other aliases.

**Comment by patriksvensson** [comment]
@phenixdotnet Thanks for this contribution! I've left some comments for you.

Could you also create an issue for this?

Thanks!

**Comment by phenixdotnet** [comment]
@patriksvensson Sure I will

**Comment by gep13** [comment]
@phenixdotnet would it be possible for you to merge this into a single commit, and re-push?  Thanks!

**Comment by phenixdotnet** [comment]
@gep13 Sure, I will do that tomorrow

**Comment by gep13** [comment]
@phenixdotnet perfect! :+1:

**Comment by phenixdotnet** [comment]
Done :)

**Comment by devlead** [comment]
@phenixdotnet branch is out-of-date with develop, could you please rebase against latest develop.

**Comment by devlead** [comment]
@phenixdotnet just checking for our release planning is this something you'll be able to look at in near future or should we postpone it to next release?

**Comment by phenixdotnet** [comment]
@devlead Branch is up to date with develop so technically you can push it to your next release. The only things I don't know how to proceed is rebasing correctly so many changes. If you have any tips or suggestions I will be very happy to learn and doing it ASAP.

Thanks !

**Comment by devlead** [comment]
@phenixdotnet This is usually my workflow

First if you haven't already add remote to official cake upstream repo (you only need to do this step once locally

`c:\temp\cake [opencover ≡]> git remote add upstream https://github.com/cake-build/cake.git`

Then my workflow is to always keep local develop branch in sync with Cake offical and only work in feature branches in my forks.

Keeping develop in sync can be done with this sequence of commands (i have it in a powerhsell file as it works on any repo with develop branch that has a upstream remote)
```
C:\temp\cake [develop ≡]> git checkout develop
C:\temp\cake [develop ≡]> git pull
C:\temp\cake [develop ≡]> git fetch upstream -v
C:\temp\cake [develop ≡]> git reset --hard upstream/develop
C:\temp\cake [develop ≡]> git push -f origin develop
```

Your develop branch last commit before this was
```
commit d452cfe227b8df7bea8206c0f7fb6cf2cc1917a6
Merge: 703c1f9 594e8a4
Author: Mattias Karlsson <matkar@wcom.se>
Date:   Wed Apr 20 11:26:12 2016 +0200
```

After resetting to upstream it's now
```
commit a0a17ca98316e15a7866ccd4d8e34f57c3e4a4ad
Merge: 1ca16c6 4a49527
Author: Patrik Svensson <patrik.svensson@tobii.com>
Date:   Fri May 13 13:43:02 2016 +0200
```

To keep my pr/feature branch in line with develop I just type, when things change in the main repo development branch
```
c:\temp\cake [opencover ≡]> git rebase develop
```
and as long as there's no conflicts it'll just rewind your branch, replay develop changes and last replay your changes. If there's a conflict you will have to resolve that as any merge conflict.

All your merge commits will now be removed and in your case you will just end up with one commit.
To update your fork you'll likely have to force push i.e.:
```
c:\temp\cake [opencover ↕]> git push -f origin develop
```

Everything should now be rebased and PR ready to be reviewed and merged.

If you want to play it safe you could do as I did now when i tested against your fork, just clone to a new foler i.e. c:\temp and work in thatm if anything goes south you'll always have an untouch complete history in your other local repo.

**Comment by phenixdotnet** [comment]
@devlead Thanks you very much, it's very appreciated !.
Some question about your workflow:
As long as you don't develop on "develop" branch do you really need those steps ? 
```
C:\temp\cake [develop ≡]> git reset --hard upstream/develop
C:\temp\cake [develop ≡]> git push -f origin develop
```

Regarding this PR, the GitHub history seem to be clean, do I need to rebase again to combine the last 2 commits ?

Again thanks you to take time for long explanations

**Comment by devlead** [comment]
It's a generic script, if you do nothing in develop you get away with just pulling the latest changes into develop.

It depends you could just push what you got and see 😉 
Interactive rebasing lets you merge commits together.

Basically `git rebase -i [last upstream commit id]` and then you can fixup or squash into a single commit.

**Comment by devlead** [comment]
@phenixdotnet merged 👍 , thanks for contributing!

**Comment by phenixdotnet** [comment]
@devlead Thanks you for your help and for the explanation about git rebase.
It's a pleasure to contribute with great projects and people like you :)

**Pull request merged**